src/braket/emulation/emulator.py

```diff
@@ -128,23 +129,12 @@ def transform(
             TaskSpecification: A compiled program with a noise model applied, if one
             exists for this emulator and apply_noise_model is true.
         """
-
         program = self._pass_manager.transform(task_specification)
 
-        # Apply measurement manually if the circuit has no measurement and no result type.
-        # This ensures that the noise model can apply readout error to the circuit, since
-        # the readout error is applied if and only if there is measurement or result type
-        # in the circuit. The measurement operations should be added even if apply_noise_model
-        # is False.
-        has_measurement = any(
-            isinstance(instr.operator, Measure) for instr in task_specification.instructions
-        )
-        if (not has_measurement) and len(task_specification.result_types) == 0:
-            task_specification.measure(target_qubits=task_specification.qubits)
-
-        return (
-            self._noise_model.apply(program) if apply_noise_model and self.noise_model else program
-        )
+        modifier = PassManager([MeasurementTransformation()])
+        if apply_noise_model:
+            modifier.append(NoiseModelTransformation(noise_model=self.noise_model))
+        return modifier.transform(program)
```

Comment:
Can we add "VerbatimTransformation()" in front of this line, so that "modifier.transform(program)" removes the verbatim box?